addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -66,5 +77,23 @@ weSnippetEditor.Class.include({
             tab: this.tabs.THEME,
         });
     },
+    /**
+     * @private
+     * @param {OdooEvent} ev
+     */
+    _onGetCustomColors: function (ev) {
+        ev.data.onSuccess([
+            this.style.getPropertyValue('--menu').trim(),
+            this.style.getPropertyValue('--footer').trim(),
+            this.style.getPropertyValue('--body').trim(),
+            this.style.getPropertyValue('--text').trim(),
+            this.style.getPropertyValue('--h1').trim(),
+            this.style.getPropertyValue('--h2').trim(),
+            this.style.getPropertyValue('--h3').trim(),
+            this.style.getPropertyValue('--h4').trim(),
+            this.style.getPropertyValue('--h5').trim(),
+            this.style.getPropertyValue('--h6').trim(),
```

Comment:
Maybe only menu, footer, body? If someone choose gray variations for its titles I really would not want them all to appear in each colorpicker.

Also, maybe use
```javascript
const variables = ['menu', 'footer', /* ... */];
variables.map(v => this.style.getPropertyValue(`--${v}`).trim());
```
?